### ISSUE TITLE
MQTTv5 support in rumqttd

### ISSUE THREAD

**Issue by swanandx** [bytebeamio/rumqtt#621]
This issue is for tracking the progress of MQTTv5 support in our broker ( rumqttd ) :rocket:  

We encourage everyone to try out MQTTv5 and help us with your valuable feedback & contributions. Feel free to comment below <3

--- 

# Features 

### WIP / Under Review

- [ ] Topic Alias ( #616 )
- [ ] Message Expiry ( #616 )

### TODO

- [ ] Shared Subscriptions
- [ ] Subscription IDs
- [ ] Subscription Options
- [ ] Will delay
- [ ] Session Expiry
- [ ] Server Disconnect
- [ ] Server reference / redirect
- [ ] Server Keep Alive
- [ ] Optional Server feature availability
- [ ] User Properties
- [ ] Flow Control
- [ ] Maximum Packet Size
- [ ] Request / Response pattern
- [ ] Payload format and content-type
- [ ] Reason code and string on ACKs
- [ ] Enhanced Auth
- [ ] Assign client ID

---

> **NOTE**: Features having _partial support_ aren't marked as Done, like request-response, server disconnect etc.  e.g. we _have_ ability to send Disconnect packets, but we aren't sending them everywhere _yet_. 
> Support for some features can be very subjective, e.g. User Properties or Enhanced Auth.

**Comment by swanandx** [comment]
Hey @tmbull, as you mentioned already, maintainers' time is the blocker, and I think getting those PRs reviewed before merging would be better, but if its urgent, we can try to figure out some workaround like merging without review ( though I don't support it much :rofl:  ) or prioritize reviews.

Can you please let me know specifically which MQTTv5 features are required by those clients, so those can be prioritized?

Thank you for your interest :100:

**Comment by tmbull** [comment]
Thanks for the quick reply! Shared subscriptions are the big feature for me, which you already have merged in. Session expiry and max packet size would be nice to have, but it is not an urgent matter so please take your time!

**Comment by dquirl-R1** [comment]
Hi all. Given that a lot of time (over 2 years) has passed since the last comment, is it still review time that improved v5 support is blocked on (CC @swanandx)? The project overall seems to have been moving along during that time, but judging by [this comment](https://github.com/bytebeamio/rumqtt/issues/972#issuecomment-2977510485) it seems like bandwidth is still very limited.

From the outside, I'm finding it difficult to judge what's the expected path forward for bigger-but-important features like this one. Are folks supposed to just wait until someone finds time? Is it @swanandx merging the PRs themselves, as they've been reviewing and merging other PRs as a maintainer? @tekjar maybe you can expand a little on how you are looking at this internally (feel free to move the discussion somewhere else if you don't wanna hijack this specific issue) in terms of being able to make progress on stuff like this at some point 🤔
